token/program-2022/src/instruction.rs

```diff
@@ -251,6 +256,10 @@ pub enum TokenInstruction {
     /// decimals value is checked by the caller.  This may be useful when
     /// creating transactions offline or within a hardware wallet.
     ///
+    /// If the accounts are `AccountWithTransferFee`s, the fee is withheld in the
```

Comment:
A `TransferCheckedWithFee` seems needed where the caller can declare the fee they expect to pay upfront.  Otherwise you could get rugged by the fee authority using `SetTransferFee` to dial the fee up to 100% on you unexpectedly

Reply:
Ah yes, that's a great idea